appenders/bash/logger.sh

```diff
@@ -1,52 +1,59 @@
 #!/bin/bash
 # Dependencies: curl
 # Usage:
+# ```
+# source logger.sh
+# pulse_debug "debug message"
+# pulse_info "info message"
+# pulse_warn "warn message"
+# pulse_trace "trace message"
+# ```
 # Export the variables: PULSE_COLLECTOR_HOST and PULSE_COLLECTOR_PORT for the host and port caontaing the Pulse
 # Log Collector
 # ```
-# export PULSE_COLLECTOR_HOST=host1.com
-# export PULSE_COLLECTOR_PORT=9004
+# export PULSE_LOG_COLLECTOR_HOST=valhalla.phdata.io
+# export PULSE_LOG_COLLECTOR_PORT=9004
+# export APPLICATION_NAME=pulse_test_default
 # ```
-# Then source this scrip in your own bash scrip
-# ``` 
-# source logger.sh
-# logger --category=category --level=level --message=message --threadname=threadname --application=application-name
-# ```
-set -euo pipefail
+
+declare ERROR_MESSAGE="Error! Message info is only possible for its argument."
 
 logger(){
-  if [ "$#" -lt 1 ]
+  # $1 = Level Info
+  # $2 = the number of funtion parameters
+  # $3 = Message to be logged
+  # $4 = Pulse Logging text
+  if [ -z "$APPLICATION_NAME" ] || [ -z "$PULSE_LOG_COLLECTOR_HOST" ] || [ -z "$PULSE_LOG_COLLECTOR_PORT" ]
   then
-  echo "Usage Info: logger --category=category --level=level --message=message --threadname=threadname --application=application-name"
-  exit 1
+    echo "Error! Env variable should have PULSE_LOG_COLLECTOR_HOST, PULSE_LOG_COLLETOR_PORT, and APPLICATION_NAME"
+  else
+    if [ "$2" -ne 1 ]
```

Comment:
Why is this comparing to `1`?

Reply:
@afoerster , `$2` is the number of `pulse_debug` function arguments. `pulse_debug` function can only have one argument, the message to be logged.